packages/snaps-execution-environments/src/common/endowments/date.ts

```diff
@@ -0,0 +1,54 @@
+import { rootRealmGlobal } from '../globalObject';
+
+const MAXIMUM_NOISE = 5;
+
+/**
+ * Creates a {@link Date} constructor, with most of the same properties as the global object.
+ * The Date.now() function has added noise as to limit its precision and prevent potential timing attacks.
+ * The Date constructor uses this now() function to seed itself if no arguments are given to the constructor.
+ *
+ * @returns A modified {@link Date} constructor with limited precision.
+ */
+function createDate() {
+  const keys = Object.getOwnPropertyNames(
+    rootRealmGlobal.Date,
+  ) as (keyof typeof Date)[];
+
+  let currentTime = 0;
+  const now = () => {
+    const actual = rootRealmGlobal.Date.now();
+    const noise =
+      rootRealmGlobal.crypto.getRandomValues(new Uint32Array(1))[0] %
```

Comment:
One concern about using good randomness for noise - this could be called significantly many times. It might be paranoya speaking, but could we use up too much entropy with this and make our randomness where we actually need it for cryptography be less random?

My original proposal was to use the dummy Math.random() here along with rounding so that chunks of time are  different but the predictability of Math.random() is not exposed.
When the `newTime>currentTime` condition is used, you don't know which random values you're getting.

Reply:
I'm fine with that as long as we are comfortable with the timing only being off by maximum 1ms. @naugtur has convinced me of this in Slack 😅